sycl/include/sycl/ext/intel/experimental/esimd/simd.hpp

```diff
@@ -45,25 +45,22 @@ template <typename Ty, int N> class simd {
   /// The number of elements in this simd object.
   static constexpr int length = N;
 
-  // TODO @rolandschulz
-  // Provide examples why constexpr is needed here.
-  //
   /// @{
   /// Constructors.
-  constexpr simd() = default;
-  template <typename SrcTy> constexpr simd(const simd<SrcTy, N> &other) {
+  simd() = default;
+  template <typename SrcTy> simd(const simd<SrcTy, N> &other) {
     if constexpr (std::is_same<SrcTy, Ty>::value)
       set(other.data());
     else
       set(__builtin_convertvector(other.data(), detail::vector_type_t<Ty, N>));
   }
```

Comment:
Not something you changed as part of this PR, but something I hope can be fixed (maybe as part of a different PR).

This constructor uses bad practice of how to deal with a templated constructor which may or may not convert. The problem with this way is that it messes with traits like `is_trivially_constructible`. Rather the converting and non-converting should be different overloads. This should be done as:

```suggestion
  simd(const simd&) = default;
  template <typename SrcTy> simd(const simd<SrcTy, N> &other) {
      set(__builtin_convertvector(other.data(), detail::vector_type_t<Ty, N>));
  }
```
The non-templated is picked over the templated and will therefore be used in the non-converting case. And because it is defaulted, things like traits work correctly.